src/main/java/org/jenkinsci/plugins/gitlab/GitlabMergeRequestWrapper.java

```diff
@@ -349,6 +350,22 @@ public GitlabNote createNote(String message, boolean shouldClose, boolean should
 
     }
 
+    public GitlabCommitStatus createCommitStatus(String status, String targetUrl) {
+        try {
+            GitlabAPI api = builder.getGitlab().get();
+            GitlabMergeRequest mergeRequest = api.getMergeRequest(project, id);
+            GitlabCommit latestCommit = getLatestCommit(mergeRequest, api);
+
+            if (latestCommit != null) {
+                return api.createCommitStatus(project, latestCommit.getId(), status, mergeRequest.getSourceBranch(), "Jenkins", targetUrl, null);
```

Comment:
@timols Do you have insight into the best way to do this? I looked into getting the data from the change set of the `AbstractBuild` we are passed in, however that involves a lot of casting and some reflection because we don't know that the changes are of type `GitChangeSet` from the git plugin. The change set it also unreliable as there will be no changes if the tests are rerun without new commits being added.

I also looked into storing the commit has in the `GitlabCause` cause, however there doesn't seem to be an easy way to get to the latest commit hash from there without using the API in the `GitlabBuild` class, which is something it currently doesn't do.

What would you recommend as the best approach to this?